packages/iov-tendermint-rpc/src/rpcclient.ts

```diff
@@ -185,16 +185,31 @@ export class WebsocketClient implements RpcStreamingClient {
       // only one of the two listeners should fire, and it will
       // deregister the other so as not to cause a leak.
 
+      const good = {
+        eventName: id,
+        listener: (data: JsonRpcResponse) => {
+          unregisterGoodAndBad();
```

Comment:
maybe just `unregisterBad()`? `unregisterGood` would already happen via `once`

Reply:
You mean having two different cleanup functions? Right now the idea is to cleanup both such that only one function is needed.